crates/node/builder/src/components/builder.rs

```diff
@@ -296,12 +297,26 @@ impl<Node, PoolB, PayloadB, NetworkB, ExecB, ConsB> NodeComponentsBuilder<Node>
 where
     Node: FullNodeTypes,
     PoolB: PoolBuilder<Node>,
-    NetworkB: NetworkBuilder<Node, PoolB::Pool>,
+    NetworkB: NetworkBuilder<
+        Node,
+        PoolB::Pool,
+        Primitives: NetworkPrimitives<
+            BlockHeader = HeaderTy<Node::Types>,
+            BlockBody = BodyTy<Node::Types>,
+        >,
```

Comment:
should this also enforce the transaction bound with pool?

Reply:
I think that would be possible and we should do it

Reply:
Just added a bound that requires the `Pooled` variant to be equal, which I think makes sense. I'm not sure about other bounds